test/runtests.jl

```diff
@@ -2,6 +2,29 @@ using BSON
 using Test
 
 roundtrip_equal(x) = BSON.roundtrip(x) == x
+function roundtrip_equal(x::AbstractArray)
```

Comment:
Special casing `roundtrip_equal` for just these couple of types seems like it could lead to confusion. I think it'd be better to define an `undef_equal(x, y)` for arrays and explicitly test `undef_equal(BSON.roundtrip(x), x)` or similar.

Reply:
Bump @Codyk12 can you address this comment?